lib/srv/db/cloud/watchers/rds_proxy.go

```diff
@@ -0,0 +1,213 @@
+/*
+Copyright 2022 Gravitational, Inc.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package watchers
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/rds"
+	"github.com/aws/aws-sdk-go/service/rds/rdsiface"
+	"github.com/gravitational/trace"
+	"github.com/sirupsen/logrus"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/services"
+	"github.com/gravitational/teleport/lib/srv/db/common"
+)
+
+// rdsDBProxyFetcher retrieves RDS Proxies and their custom endpoints.
+type rdsDBProxyFetcher struct {
+	cfg rdsFetcherConfig
+	log logrus.FieldLogger
+}
+
+// newRDSDBProxyFetcher returns a new RDS Proxy fetcher instance.
+func newRDSDBProxyFetcher(config rdsFetcherConfig) (Fetcher, error) {
+	if err := config.CheckAndSetDefaults(); err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return &rdsDBProxyFetcher{
+		cfg: config,
+		log: logrus.WithFields(logrus.Fields{
+			trace.Component: "watch:rdsproxy",
+			"labels":        config.Labels,
+			"region":        config.Region,
+		}),
+	}, nil
+}
+
+// Get returns RDS Proxies and proxy endpoints matching the watcher's
+// selectors.
+func (f *rdsDBProxyFetcher) Get(ctx context.Context) (types.Databases, error) {
+	databases, err := f.getRDSProxyDatabases(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return filterDatabasesByLabels(databases, f.cfg.Labels, f.log), nil
+}
+
+// getRDSProxyDatabases returns a list of database resources representing RDS
+// Proxies and custom endpoints.
+func (f *rdsDBProxyFetcher) getRDSProxyDatabases(ctx context.Context) (types.Databases, error) {
+	// Get a list of all RDS Proxies. Each RDS Proxy has one "default"
+	// endpoint.
+	rdsProxies, err := getRDSProxies(ctx, f.cfg.RDS, common.MaxPages)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	// Get all RDS Proxy custom endpoints sorted by the name of the RDS Proxy
+	// that owns the custom endpoints.
+	customEndpointsByProxyName, err := getRDSProxyCustomEndpoints(ctx, f.cfg.RDS, common.MaxPages)
+	if err != nil {
+		f.log.Debugf("Failed to get RDS Proxy endpoints: %v.", err)
+	}
+
+	databases := types.Databases{}
```

Comment:
nit: [CodeReviewComments#declaring-empty-slices](https://github.com/golang/go/wiki/CodeReviewComments#declaring-empty-slices)

```suggestion
	var databases types.Databases
```